sotodlib/io/hk_utils.py

```diff
@@ -273,8 +436,17 @@ def get_detcosamp_hkaman(config, det_aman):
     """
     start = float(det_aman.timestamps[0])
     stop = float(det_aman.timestamps[-1])
-    data = get_grouped_hkdata(start, stop, config)
 
-    hkamans_detcosamp = make_hkaman(data, det_cosampled=True, det_aman=det_aman)
-    return hkamans_detcosamp
-# TODO: insert a progress bar for get_grouped_hkdata
+    if config is not None:
+        data = sort_hkdata_fromconfig(start, stop, config)
+        amans = make_hkaman(data, det_aman=det_aman, alias_exists=True, det_cosampled=True)
+        return amans
+    elif fields is not None:
+        if alias is None:
+            data = sort_hkdata(start, stop, fields, data_dir)
+            amans = make_hkaman(data, det_cosampled=True, det_aman=det_aman)
```

Comment:
similarly to other comment. Functions like `make_hkaman`, I wouldn't depend on a default kwarg parameter, if `alias_exists` needs to be false then explicitly set it to false here.